src/app/src/test/kotlin/org/imperial/mrc/hint/integration/adr/ADRClientTests.kt

```diff
@@ -41,24 +41,24 @@ class ADRClientTests
     {
         val sut = ADRFuelClient(ConfiguredAppProperties(), "fakekey")
         val response = sut.get("garbage")
-        assertThat(response.statusCodeValue).isEqualTo(500)
+        assertThat(response.statusCodeValue).isEqualTo(400)
         val errors = ObjectMapper().readValue<JsonNode>(response.body!!)["errors"]
-        assertThat(errors.isArray).isTrue()
+        assertThat(errors.isArray).isTrue
         assertThat(errors.count()).isEqualTo(1)
         assertThat(errors[0]["error"].textValue()).isEqualTo("OTHER_ERROR")
-        assertThat(errors[0]["detail"].textValue()).isEqualTo("Could not parse response.")
+        assertThat(errors[0]["detail"].textValue()).isEqualTo("Bad request - Action name not known: garbage")
     }
 
     @Test
     fun `returns an error when uploading a file to a non-existent endpoint`()
     {
         val sut = ADRFuelClient(ConfiguredAppProperties(), "fakekey")
         val response = sut.postFile("garbage", listOf(), Pair("garbage", File("/dev/null")))
-        assertThat(response.statusCodeValue).isEqualTo(500)
+        assertThat(response.statusCodeValue).isEqualTo(400)
         val errors = ObjectMapper().readValue<JsonNode>(response.body!!)["errors"]
-        assertThat(errors.isArray).isTrue()
+        assertThat(errors.isArray).isTrue
         assertThat(errors.count()).isEqualTo(1)
         assertThat(errors[0]["error"].textValue()).isEqualTo("OTHER_ERROR")
-        assertThat(errors[0]["detail"].textValue()).isEqualTo("Could not parse response.")
+        assertThat(errors[0]["detail"].textValue()).isEqualTo("Bad request - Action name not known: garbage")
```

Comment:
The change to `asResponseEntity` was really worthwhile even independently of the current ticket, it's a big improvement to be getting informative error messages back here!

Reply:
Agreed ☝️